### PR TITLE
Split out list

### PR DESCRIPTION
**Pull request by chrisnojima**
This splits the list into its own connected component which really reduced what the root connector needs to do. 
Also in order to render this thing it needed to go through multiple levels of hocs which wasn't really necessary

@keybase/react-hackers

### CODE REVIEW

File: shared/chat/conversation/banner/container.js.flow [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// @flow
+import {Component} from 'react'
+export type OwnProps = {}
+
+export default class BannerContainer extends Component<void, OwnProps, void> {}
```
Comment (chrisnojima) [inline shared/chat/conversation/banner/container.js.flow]: Help ensure OwnProps are passed down correctly

File: shared/chat/conversation/container.js [hunk 2]
```diff
@@ -1,170 +1,52 @@
...
-import {onUserClick} from '../../actions/profile'
-import {openDialog as openRekeyDialog} from '../../actions/unlock-folders'
-import {pick} from 'lodash'
-import {withState, withHandlers, compose} from 'recompose'
+import {withState, withHandlers, compose, branch, renderNothing, lifecycle} from 'recompose'
 
-import type {TypedState} from '../../constants/reducer'
-import type {OpenInFileUI} from '../../constants/kbfs'
 import type {Props} from '.'
-
```
Comment (chrisnojima) [inline shared/chat/conversation/container.js]: 🔥

File: shared/chat/conversation/container.js [hunk 3]
```diff
@@ -2,169 +2,77 @@
 import * as Constants from '../../constants/chat'
 import * as Creators from '../../actions/chat/creators'
 import Conversation from './index'
-import HiddenString from '../../util/hidden-string'
```
Comment (chrisnojima) [inline shared/chat/conversation/container.js]: 🔥

File: shared/chat/conversation/container.js [hunk 4]
```diff
@@ -2,169 +2,77 @@
...
+type StateProps = {
+  finalizeInfo: ?Constants.FinalizeInfo,
   listScrollDownCounter: number,
-  onEditLastMessage: () => void,
-  onFocus: () => void,
-  onScrollDown: () => void,
-} & Props
-
-class ConversationContainer extends Component<void, ConversationContainerProps, void> {
-  componentWillReceiveProps (nextProps: Props) {
```
Comment (chrisnojima) [inline shared/chat/conversation/container.js]: handle this in recompose

File: shared/chat/conversation/container.js [hunk 5]
```diff
@@ -2,169 +2,77 @@
...
-        selectedConversationIDKey,
-        validated: selected && selected.state === 'unboxed',
-        you,
-        supersedes,
-        supersededBy,
-        threadLoadedOffline: conversationState.loadedOffline,
-        finalizeInfo,
-        editingMessage: state.chat.get('editingMessage'),
-      }
+      participants = selected && selected.participants || List()
```
Comment (chrisnojima) [inline shared/chat/conversation/container.js]: this stuff could go away if we changed the dependency on OldProfileResetNotice

File: shared/chat/conversation/index-hoc.js [hunk 6]
```diff
@@ -1,96 +0,0 @@
-// @flow
```
Comment (chrisnojima) [inline shared/chat/conversation/index-hoc.js]: 🔥

File: shared/chat/conversation/index.js.flow [hunk 7]
```diff
@@ -1,50 +1,26 @@
 // @flow
 import {Component} from 'react'
-import {List} from 'immutable'
 import * as Constants from '../../constants/chat'
 
 export type Props = {
-  focusInputCounter: number,
+  editLastMessageCounter: number,
   finalizeInfo: ?Constants.FinalizeInfo,
-  firstNewMessageID: ?Constants.MessageID,
```
Comment (chrisnojima) [inline shared/chat/conversation/index.js.flow]: 🔥

File: shared/chat/conversation/index.native.js [hunk 8]
```diff
@@ -18,7 +17,15 @@ import type {Props} from './index'
 const Conversation = (props: Props) => (
   <Box style={{...globalStyles.flexBoxColumn, flex: 1}}>
     <Header sidePanelOpen={props.sidePanelOpen} onToggleSidePanel={props.onToggleSidePanel} onBack={props.onBack} />
-    <List {...props.listProps} />
+    <List
```
Comment (chrisnojima) [inline shared/chat/conversation/index.native.js]: lets be explicit about all our props

File: shared/chat/conversation/list-hoc.js [hunk 9]
```diff
@@ -1,66 +0,0 @@
-// @flow
```
Comment (chrisnojima) [inline shared/chat/conversation/list-hoc.js]: 🔥

File: shared/chat/conversation/list/container.js [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+    selectedConversation: selectedConversationIDKey,
+    supersededBy,
+    supersedes,
+    validated,
+    you,
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
+  _onLoadAttachment: (selectedConversation, messageID, filename) => { dispatch(Creators.loadAttachment(selectedConversation, messageID, downloadFilePath(filename), false, false)) },
```
Comment (chrisnojima) [inline shared/chat/conversation/list/container.js]: flow really got confused by the pattern of overwriting a prop with a value of the same name. So i renamed these things to use _On... instead. I think this is also good cause it tells the devs these are meant to be overridden and not used directly

File: shared/chat/conversation/list/container.js [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+      username: resetUser,
+    }
+    return messages.unshift(supersedesMessage)
+  }
+
+  return messages
+}
+
+// TODO remove this. Not needed w/ connected messages
+// $FlowIssue wants props with optionsFn inside of it, which this thing itself is building. This'll all go away soon
```
Comment (chrisnojima) [inline shared/chat/conversation/list/container.js]: annoyingly even though it never references props.optionsFn, flow complains that its missing (which it is, cause this is the thing that's adding it). this'll all go away soon

File: shared/chat/conversation/container.js (hunk 5 above)
Comment (MarcoPolo) [inline shared/chat/conversation/container.js]: Change how?

Comment (chrisnojima) [inline shared/chat/conversation/container.js]: Make it connected also

File: shared/chat/conversation/list/container.js [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+  if (selectedConversationIDKey && Constants.isPendingConversationIDKey(selectedConversationIDKey)) {
+    const tlfName = Constants.pendingConversationIDKeyToTlfName(selectedConversationIDKey)
+    if (tlfName) {
+      participants = List(tlfName.split(','))
+      followingMap = pick(origFollowingMap, participants.toArray())
+      metaDataMap = origMetaDataMap.filter((k, v) => participants.contains(v))
+      validated = true
+    }
+  } else if (selectedConversationIDKey && selectedConversationIDKey !== Constants.nothingSelected) {
+    const conversationState = state.chat.get('conversationStates').get(selectedConversationIDKey)
```
Comment (MarcoPolo) [inline shared/chat/conversation/list/container.js]: 🐑 these could be selectors too, right?

File: shared/chat/conversation/container.js [hunk 13]
```diff
@@ -2,169 +2,77 @@
...
+      rekeyInfo = state.chat.get('rekeyInfos').get(selectedConversationIDKey)
+      finalizeInfo = state.chat.get('finalizedState').get(selectedConversationIDKey)
+      supersedes = Constants.convSupersedesInfo(selectedConversationIDKey, state.chat)
+      supersededBy = Constants.convSupersededByInfo(selectedConversationIDKey, state.chat)
```
Comment (MarcoPolo) [inline shared/chat/conversation/container.js]: Hmm so this has to be done twice for 2 different containers? Maybe it's worth connecting this now if it's not too much work

File: shared/chat/conversation/list/container.js (hunk 10 above)
Comment (MarcoPolo) [inline shared/chat/conversation/list/container.js]: 👍 I've done this elsewhere and agree it's better

File: shared/chat/conversation/container.js (hunk 13 above)
Comment (MarcoPolo) [inline shared/chat/conversation/container.js]: okay to do in a sep. pr

File: shared/chat/conversation/list/container.js (hunk 12 above)
Comment (chrisnojima) [inline shared/chat/conversation/list/container.js]: yeah, once i have this stuff all connected, we can move to entities and conversationStates will go away

File: shared/chat/conversation/container.js (hunk 13 above)
Comment (chrisnojima) [inline shared/chat/conversation/container.js]: this'll go away when the messages get connected i think

**Pull request merged**